temperature/utils.py

```diff
@@ -0,0 +1,39 @@
+import os
+
+import httpx
+from fastapi import Depends, HTTPException
+from sqlalchemy.ext.asyncio import AsyncSession
+from dotenv import load_dotenv
+
+from weather.crud import get_all_cities
+from dependencies import get_db
+
+load_dotenv()
+
+API_URL = os.getenv("API_URL")
+API_KEY = os.getenv("API_KEY")
+
+
+async def check_service_connection():
+    async with httpx.AsyncClient() as client:
+        response = await client.get(
+            f"{API_URL}/current.json", params={"key": API_KEY, "q": "London"}
+        )
+    if response.status_code != 200:
+        error_detail = response.json() if response.headers.get("Content-Type") == "application/json" else response.text
+        raise HTTPException(status_code=response.status_code, detail=error_detail)
```

Comment:
The `HTTPException` is raised with the status code from the external API response. Ensure that this behavior is intended and that the application handles these exceptions appropriately.